internal/framework/provider/config.go

```diff
@@ -0,0 +1,103 @@
+package provider
```

Comment:
Right, as discussed over Slack https://github.com/hashicorp/terraform-provider-scaffolding-framework is the template TF shares for provider structure based in the Framework.
For us, since we currently have both framework based and SDKv2 based resources in the provider, added Framework to distinguish easily.
Also, for some other providers in the middle of migration, I have noticed mostly to adhere to Go project structure as well, this is how they are structured:
```
internal/framework/
     internal/framework/provider
     internal/framework/resources
internal/sdkv2provider/
     internal/sdkv2provider/provider
     internal/sdkv2provider/resources
```
However, above would require us to move our current resources into internal as well, which I am guessing would be 1-2 days work max, probably less?

@andreaangiolillo Let me know your thoughts on this, no strong opinions from me, as we could take this up as one of the goals of this migration.